security/pkg/nodeagent/sds/monitoring.go

```diff
@@ -15,150 +15,43 @@
 package sds
 
 import (
-	"github.com/prometheus/client_golang/prometheus"
+	"istio.io/pkg/monitoring"
 )
 
+// TODO: Create an endpoint in Citadel agent to expose metrics.
+// Previously exposed metrics: pending_push_per_connection, stale_conn_count_per_connection,
+// pushes_per_connection, push_errors_per_connection, pushed_root_cert_expiry_timestamp, pushed_server_cert_expiry_timestamp
 var (
 	// totalPushCounts records total number of SDS pushes since server starts serving.
-	totalPushCounts = prometheus.NewCounter(prometheus.CounterOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "total_pushes",
-		Help:      "The total number of SDS pushes.",
-	})
+	totalPushCounts = monitoring.NewSum(
+		"total_pushes",
```

Comment:
before wasnt the metric citadel-agent_sds_service_total_pushes and now its just total_pushes? or does the prefix get added somewhere I am missing? If it shows up in prometheus as just `total_pushes` that won't be very useful

Reply:
Not sure about this at Prometheus backend level (we haven't exported the data to prometheus backend yet). This is okay today since we're only looking at citadel agent `/metrics` endpoint, but maybe you're right and we need to add some prefix. I'll add it soon

Reply:
we definitely should get prometheus scraping set up - should be pretty simple I think (although I haven't looked at the config too much)

Reply:
I think we can actually add the prefix in the namespace when we export a new Prometheus endpoint. I'll do that as a separate PR for both Citadel and Citadel Agent.